lua/terrortown/menus/gamemode/appearance/hudswitcher.lua

```diff
@@ -0,0 +1,166 @@
+			default = data.default,
+			OnChange = function(_, value)
+				value = value or false
+				currentHUD[key] = value
+
+				if isfunction(data.OnChange) then
+					data.OnChange(currentHUD, value)
+				end
+			end
+		})
+	end
+}
+
+local function PopulateHUDSwitcherPanelSettings(parent, currentHUD)
+	parent:Clear()
+
+	parent:MakeHelp({
+		label = "help_hud_special_settings"
+	})
+
+	for key, data in pairs(currentHUD:GetSavingKeys() or {}) do
+		hudSwicherSettings[data.typ](parent, currentHUD, key, data)
+	end
+end
+
+CLGAMEMODESUBMENU.base = "base_gamemodesubmenu"
+
+CLGAMEMODESUBMENU.priority = 99
+CLGAMEMODESUBMENU.title = "submenu_appearance_hudswitcher_title"
+
+function CLGAMEMODESUBMENU:Populate(parent)
+	local form = vgui.CreateTTT2Form(parent, "header_hud_select")
+
+	local currentHUDName = HUDManager.GetHUD()
+	local currentHUD = huds.GetStored(currentHUDName)
+	local hudList = huds.GetList()
+	local restrictedHUDs = ttt2net.GetGlobal({"hud_manager", "restrictedHUDs"})
+	local forcedHUD = ttt2net.GetGlobal({"hud_manager", "forcedHUD"})
+	local validHUDs = {}
+
+	if not currentHUD.GetSavingKeys then
+		form:MakeHelp({
+			label = "help_hud_game_reload"
+		})
+
+		return
+	end
+
+	if forcedHUD then
+		validHUDs[1] = forcedHUD
+	else
+		for i = 1, #hudList do
+			local hud = hudList[i]
+
+			-- do not add HUD to the selection list if restricted
+			if table.HasValue(restrictedHUDs, hud.id) then continue end
+
+			validHUDs[#validHUDs + 1] = hud.id
+		end
+	end
+
+	form:MakeComboBox({
+		label = "label_hud_select",
+		choices = validHUDs,
+		selectName = currentHUDName,
+		OnChange = function(_, _, value)
+			HUDManager.SetHUD(value)
+		end,
+		default = ttt2net.GetGlobal({"hud_manager", "defaultHUD"})
+	})
+
+	PopulateHUDSwitcherPanelSettings(vgui.CreateTTT2Form(parent, "header_hud_customize"), currentHUD)
+
+	-- REGISTER UNHIDE FUNCTION TO STOP HUD EDITOR
+	HELPSCRN.menuFrame.OnShow = function(slf)
+		if HELPSCRN:GetOpenMenu() ~= "appearance_hudswitcher" then return end
+
+		HUDEditor.StopEditHUD()
+	end
+end
+
+function CLGAMEMODESUBMENU:PopulateButtonPanel(parent)
+	local currentHUDName = HUDManager.GetHUD()
+	local currentHUD = huds.GetStored(currentHUDName)
+
+	local buttonReset = vgui.Create("DButtonTTT2", parent)
+
+	buttonReset:SetText("button_reset")
+	buttonReset:SetSize(100, 45)
+	buttonReset:SetPos(475, 20)
+	buttonReset.DoClick = function(btn)
+		if not currentHUD then return end
+
+		currentHUD:Reset()
+		currentHUD:SaveData()
+	end
+	buttonReset:SetEnabled(not currentHUD.disableHUDEditor)
+
+	local buttonEditor = vgui.Create("DButtonTTT2", parent)
+
+	buttonEditor:SetText("button_hud_editor")
+	buttonEditor:SetSize(175, 45)
+	buttonEditor:SetPos(600, 20)
+	buttonEditor.DoClick = function(btn)
+		if not currentHUDName then return end
+
+		HUDEditor.EditHUD(currentHUDName)
+
+		HELPSCRN.menuFrame:HideFrame()
+	end
+	buttonEditor:SetEnabled(not currentHUD.disableHUDEditor)
+end
+
+function CLGAMEMODESUBMENU:HasButtonPanel()
```

Comment:
Would it be possible to just call the button panel function or is this needed as a check somewhere else? Otherwise this could be removed and other panels would just not do anything in it

Reply:
I don't know how it would be possible without executing the button panel code even though no panel is added